docs/integration_fetching_data.md

```diff
@@ -0,0 +1,146 @@
+
+With polling, we will fetch the latest data from the API at a specified interval. Your integration will then supply this data to its entity, which is written to Home Assistant.
+
+Because polling is so common, Home Assistant by default assumes that your entity is based on polling. If this is not the case, return `False` from the `should_poll` property. When you disable polling, your integration will be responsible for calling one of the methods to indicate to Home Assistant that it's time to write the entity state to Home Assistant:
+
+- `Entity.schedule_update_ha_state(update_data=False)`/`Entity.async_schedule_update_ha_state(update_data=False)` will schedule an update of the entity. If `update_data` is set to `True`, Home Assistant will call your entities update method (`update()`/`async_update()`) prior to writing the state.
+- `Entity.async_write_ha_state()` is an async callback that will write the state to the state machine within yielding to the event loop. It will not call your entity update method.
+
+# Polling API endpoints
+
+We're going to explain a few different API types ones here and the best way to integrate them in Home Assistant. Note that some integrations will encounter a combination of the ones below.
+
+## One API endpoint with all the data
+
+This API will have a single method to fetch data for all the entities that you have in Home Assistant. In this case we'll want to poll this endpoint on an interval and then let entities know as soon as new data is available.
+
+Home Assistant provides a Data Update Coordinator class to help you manage this as efficient as possible.
+
+```python
+import logging
+
+from homeassistant.helpers import debounce, entity
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator, UpdateFailed
+
+
+_LOGGER = logging.getLogger(__name__)
+
+async def create_coordinator(api):
+    async def async_update_data():
+        """Fetch data from API endpoint."""
+        try:
+            await api.fetch_data()
+        except ApiError:
+            raise UpdateFailed
+
+
+    coordinator = DataUpdateCoordinator(
+        hass,
+        _LOGGER,
+        # Name of the data. For logging purposes.
+        "sensor",
+        async_update_data,
+        # Polling interval. Will only be polled if there are subscribers.
+        timedelta(seconds=30),
+        # Debouncer to limit request_refresh
+        debounce.Debouncer(
+            hass,
+            _LOGGER,
+            # Timeout to wait between refreshes
+            0.3,
+            # When a refresh is requested, should we refresh right await and then wait
+            # Or should we wait and refresh at the end of timeout.
+            True
```

Comment:
Same as above here. Especially the comment of the last parameter will be confusing for future developers (if they keep copying that around instead of using kwargs).